packages/alchemy/src/provider.ts

```diff
@@ -145,10 +149,18 @@ export class AlchemyProvider extends SmartAccountProvider<HttpTransport> {
 
     if (this.feeOptsSet) {
       return this.withPaymasterMiddleware(
-        alchemyPaymasterAndDataMiddleware(this, config)
+        alchemyPaymasterAndDataMiddleware<
+          TAccount,
+          HttpTransport,
+          SmartAccountProvider<TAccount, HttpTransport> & { account: TAccount }
```

Comment:
I think we can update this to

```suggestion
          ConnectedSmartAccountProvider<TAccount, HttpTransport>
```